Plugin/workspace.h

```diff
@@ -174,10 +174,15 @@ class WXDLLIMPEXP_SDK clCxxWorkspace : public LocalWorkspaceCommon
     wxArrayString GetWorkspaceFolders() const;
 
     /**
-     * @brief create 'compile_commands' json object for the workspace projects (only the enabled ones)
-     * or compile_flags.txt file
+     * @brief create 'compile_commands' json object for the workspace (enabled) projects
      */
-    cJSON* CreateCompileCommandsJSON(bool createCompileFlagsTxt, wxArrayString* generated_paths) const;
+    nlohmann::json CreateCompileCommandsJSON() const;
```

Comment:
Let's add a new header somewhere (in the `ROOT/CodeLite/` folder) with the following content:

```c++
#pragmace once

#include <assistant/common/json.hpp>

using clJSON = nlohmann::json;
```

I have already "burnt" once when I exposed the JSON implementation (`cJSON`) in various places. This will help any future migrations.

WDYT?